### DISCUSSION TITLE
libjnidispatch.so not built?

### DISCUSSION THREAD

**Discussion by ar-jan** [dbeaver/dbeaver#21804] [General]
### Description

When compiling dbeaver 23.2.3 for Void Linux I run into this issue:

> => dbeaver-23.2.3_1: running post-install hook: 06-strip-and-debug-pkgs ...
>    Stripped executable: /usr/lib/dbeaver/dbeaver
> /usr/bin/strip: Unable to recognise the format of the input file `/destdir//dbeaver-23.2.3/usr/lib/dbeaver/plugins/com.sun.jna_5.13.0.v20230812-1000/com/sun/jna/sunos-sparc/libjnidispatch.so'
> => ERROR: dbeaver-23.2.3_1: failed to strip /usr/lib/dbeaver/plugins/com.sun.jna_5.13.0.v20230812-1000/com/sun/jna/sunos-sparc/libjnidispatch.so
> => ERROR: dbeaver-23.2.3_1: post-install_06-strip-and-debug-pkgs: 'find ${PKGDESTDIR} -type f' exited with 1
> => ERROR:   in hook() at common/hooks/post-install/06-strip-and-debug-pkgs.sh:68
> => ERROR:   in run_func() at common/xbps-src/shutils/common.sh:57
> => ERROR:   in run_pkg_hooks() at common/xbps-src/shutils/common.sh:298
> => ERROR:   in main() at common/xbps-src/libexec/xbps-src-doinstall.sh:66
> 

Looking in the relevant directory:

> $ file /usr/lib/dbeaver/plugins/com.sun.jna_5.13.0.v20230812-1000/com/sun/jna/sunos-sparc/libjnidispatch.so
> cannot open `/usr/lib/dbeaver/plugins/com.sun.jna_5.13.0.v20230812-1000/com/sun/jna/sunos-sparc/libjnidispatch.so' (No such file or directory)

Is this related to the switch to Eclipse 2023-09 maybe?

### DBeaver Version

Community Edition 23.2.3

### Operating System

Void Linux

### Database and driver

N/A

**Comment by chirontt** [comment]
Void Linux packages use the `musl` libc library, right?

I think packages used in the DBeaver software are compiled using the `glibc` library, not `musl`, so they can't be used (or even compiled/linked) in Void Linux system. Package such as the [JNA](https://github.com/java-native-access/jna) library, mentioned in your compile log, includes some `*.so` native libraries which were compiled with `glibc` thus can't be used in Void Linux. Even if you manage to get a JNA library specifically compiled with `musl`, there are other `*.so` native libraries in Eclipse that have the same `glibc` problem (to Void Linux), unless you can recompile Eclipse itself with `musl`. Maybe you could ask the Void Linux people if they have a `musl`-specific version of Eclipse that you can use to compile DBeaver with.

**Comment by ar-jan** [comment]
Void Linux supports musl besides glibc, but dbeaver is built [only against glibc](https://github.com/void-linux/void-packages/blob/master/srcpkgs/dbeaver/template).  I'm bumping the packaged version for Void Linux, and this is a new issue introduced with 23.2.3.
